src/FsAutoComplete.Core/FakeSupport.fs

```diff
@@ -0,0 +1,47 @@
+namespace FsAutoComplete
+
+open Fake.Runtime
+open Fake.Runtime.Trace
+
+module FakeSupport =
+  type DebugTraceListener() =
+    interface ITraceListener with
+      /// Writes the given message to the Console.
+      member this.Write msg = 
+        //let color = colorMap msg
+        match msg with
+        | ImportantMessage text | ErrorMessage text ->
+          //writeText true color true text
+          Debug.print "%s" text
+        | LogMessage(text, newLine) | TraceMessage(text, newLine) ->
+          //writeText false color newLine text
+          Debug.print "%s" text
+
+  let mutable private loggingSetup = false
+  let private setupLogging () =
+    if not loggingSetup then
+      listeners.Clear()
+      listeners.Add(DebugTraceListener())
+      loggingSetup <- true
+
+  let detectFakeScript (file) =
+    setupLogging()
+    if not (System.IO.File.Exists file) then None
+    else
+      let config = FakeRuntime.createConfigSimple Verbose [] file [] true false
+      let prepared = FakeRuntime.prepareFakeScript config
+      let isFakeScript = prepared.DependencyType <> FakeRuntime.PreparedDependencyType.DefaultDependencies
+      
+      if isFakeScript then Some (config, prepared) else None
+
+  let getProjectOptions (config) (prepared) =
+    setupLogging()
+    let prov = FakeRuntime.restoreAndCreateCachingProvider prepared
+    let context, cache = CoreCache.prepareContext config prov
+    let args = context.Config.CompileOptions.AsArgs
+    let args =
+      args |> Seq.toList
+      |> List.filter (fun arg -> arg <> "--")
+    
+    "--simpleresolution" :: "--targetprofile:netstandard" :: "--nowin32manifest" :: args
+    |> List.toArray
```

Comment:
can we discuss about it?
mostly i'd like to avoid the deps on FAKE/paket and transitive (mono.cecil) because these constraint us later.
If we can minimize the surface, better.
we can use `paket github` refs to the fake .fs, or create an ah hoc assembly
Or use it out of process, as console app, so who care about deps.

if is a script initialization thing (`paket`), i'd like to move it in DPW.
so every fsx load support the `nuget paket` syntax OOTB

Reply:
> i'd like to avoid the deps

I'm not sure if we have a problem here. FAKE is important, getting good UX in FSAC/Ionide seems really important from my point of view. 
Also, FAKE is well maintained and regularly released project so it would bottleneck us on releases, and @matthid said he'd be open to giving us release rights for the packages we need.

I'd be in favor of merging this.

Reply:
Yes I am in favor too.
There is no problem, and don’t think we’ll get stuck.

And it’s lovely to have FAKE supported ootb.

Just trying to figure out if we can somehow minimize deps.
If not, that’s it, not hard feeling.

Maybe the only thing is that because is just a `fsx -> fsharpprojectiption` I’d like to move that logic in DPW, so:
- is easier to test
- works if anyone want to get the options for a fake script

I need anyway to complicate it a bit the fsx initalization to support in DPW some scenarios (choose tfm, `dotnet fsi` vs `fsi`)

But I can refactor our later, np